superset/db_engine_specs.py

```diff
@@ -1367,6 +1367,15 @@ class BQEngineSpec(BaseEngineSpec):
     As contributed by @mxmzdlv on issue #945"""
     engine = 'bigquery'
 
+    """
+    https://www.python.org/dev/peps/pep-0249/#arraysize
+    raw_connections bypass the pybigquery query execution context and deal with
+    raw dbapi connection directly.
+    If this value is not set, the default value is set to 1, as described here,
+    https://googlecloudplatform.github.io/google-cloud-python/latest/_modules/google/cloud/bigquery/dbapi/cursor.html#Cursor
+    """
+    arraysize = 5000
```

Comment:
I like the idea of setting the arraysize as a class attr, that would be the expected way to set it in derived classes. Maybe add those two lines in `BaseEngineSpec.fetch_data`:
```python
if cls.arraysize:
    cursor.arraysize = cls.arraysize
```
and set `arraysize = None` in the base class

Reply:
also let' s add a short comment on the `arraysize = 5000` line that says where you got that number from (the default value in `pybigquery`)